src/ert/scheduler/scheduler.py

```diff
@@ -255,8 +247,7 @@ async def _monitor_and_handle_tasks(
 
             if not self.is_active():
                 if self._ee_uri is not None:
-                    await self._events.put(CLOSE_PUBLISHER_SENTINEL)
-                    await self._publisher_done.wait()
+                    await self._events.join()
```

Comment:
I think we had this or something similar earlier, and it caused us to end before all events had been processed (in this case, the queue would be empty because the final events had not been added yet)

Reply:
`await self._events.join()` would make sure that all the events have been processed. Essentially, `CLOSE_PUBLISHER_SENTINEL` was added only here, which serves the same logic.

Reply:
I share Jonathans worry, if there is something to be sent but it it is not sent yet, `self._events()` could be empty prematurely.